src/converse-chatview.js

```diff
@@ -855,6 +856,20 @@ converse.plugins.add('converse-chatview', {
                 }
             },
 
+            async onMuteUser (message) {
+                try {
+                    await this.model.onMuteUser(message);
+                    const jid = message.get('from');
+                    const { messages } = this.model;
+                    messages.each(msg => {
+                        const from = msg.get('from');
+                        if (from === jid) msg.destroy();
+                    });
+                } catch (error) {
+                    log.error(error);
```

Comment:
Here we need to provide an error feedback to the user and then return.

See [this example](https://github.com/conversejs/converse.js/blob/0da80671290d50b8e5b9e8cac895e4966d42c81c/src/converse-chatview.js#L150) of how to show an error message  to the user.